marlowe-playground-client/src/MainFrame/State.purs

```diff
@@ -409,10 +399,7 @@ handleAction settings (HandleActusBlocklyMessage (ActusBlockly.CurrentTerms flav
         Failure e -> void $ query _actusBlocklySlot unit (ActusBlockly.SetError ("Server error! " <> (showErrorDescription (runAjaxError e).description)) unit)
         _ -> void $ query _actusBlocklySlot unit (ActusBlockly.SetError "Unknown server error!" unit)
 
--- Replicate the state of unsavedChanges from the submodule/subcomponent into the MainFrame state
-handleAction _ (HandleActusBlocklyMessage _) = do
-  hasUnsavedChanges <- queryCurrentEditorForUnsavedChanges
-  assign _hasUnsavedChanges hasUnsavedChanges
+handleAction settings (HandleActusBlocklyMessage ActusBlockly.CodeChange) = assign _hasUnsavedChanges true
```

Comment:
Could be `handleAction _ (HandleActusBlocklyMessage ActusBlockly.CodeChange) = assign _hasUnsavedChanges true`.